### ISSUE TITLE
Using QuadProgpp with Eigen?

### ISSUE THREAD

**Issue by shoshievass** [liuq/QuadProgpp#5]
Hi!

I'm hoping to use this solver with matrix algebra through the Eigen library. Has anyone given any thought to doing something like this (using an alternate matrix algebra library)? 

Do you foresee any problems with doing this by replacing references to vectors/matrices with the Eigen defs?

**Comment by liuq** [comment]
First of all, sorry for the delay in my answer. 

In principle the algorithm is agnostic with respect to the implementation of vector/matrix data structure (indeed it has been also ported to uBlas data structures). A few things should be rearranged, though (e.g., setting matrix sizes), using the right syntax for accessing elements (I see that Eigen uses a slightly different accessor). Moreover, if you want you can exploit Eigen cholesky decomposition instead of the one implemented in QuadProgpp. 

In summary, it would be a bit of work in adapting the syntax, but the outcome should work.

**Comment by MatthewPeterKelly** [comment]
+1
It would be great if this library was implemented using Eigen!

**Comment by shoshievass** [comment]
Turns out someone has already done this! https://github.com/fnoorian/quadprogpp/
